.github/workflows/generate-entity-types.yaml

```diff
@@ -0,0 +1,165 @@
+name: Generate Entity Types
+
+on:
+  # these must be duplicated :(
+  workflow_dispatch:
+    inputs:
+      repository-list:
+        description: 'Repository list to generate entity types from, space-separated. Use the same ref format as `run-config.yaml`. Leave blank to use the current list.'
+        required: false
+        default: ''
+        type: string
+      force-generate-joins:
+        description: 'If join generation should be forced (still generates joins even if not fully resolved)'
+        required: false
+        default: false
+        type: boolean
+      base-branch:
+        description: 'Branch of `mod-fqm-manager` to use as a base for the PR or `master` if empty'
+        required: false
+        default: 'master'
+        type: string
+      created-branch:
+        description: 'Branch of `mod-fqm-manager` to make the PR with; if empty, `autogenerated-staging` will be used. No effect if `create-pr` is set to `no`.'
+        required: false
+        default: 'autogenerated-staging'
+        type: string
+      create-pr:
+        description: If a PR should be created and if it should be non-draft
+        required: false
+        default: 'yes'
+        type: choice
+        options:
+          - 'yes'
+          - draft
+          - 'no'
+      send-slack:
+        description: If Slack messages should be sent alerting about the PR creation and any team's issues
+        required: false
+        default: false
+        type: boolean
+  workflow_call:
+    inputs:
+      repository-list:
+        required: false
+        default: ''
+        type: string
+      force-generate-joins:
+        required: false
+        default: false
+        type: boolean
+      base-branch:
+        required: false
+        type: string
+      created-branch:
+        required: false
+        type: string
+      create-pr:
+        required: true
+        type: string
+      send-slack:
+        description: If Slack messages should be sent alerting about the PR creation and any team's issues
+        required: true
+        type: boolean
+
+jobs:
```

Comment:
The main logic of running each step. Goes through each `scripts/entity-generation/...` in order with extra calls to create PR/etc